_posts/2020-02-26-flink-for-data-warehouse.md

```diff
@@ -0,0 +1,99 @@
+
+Apache Flink has been a proven scalable system to handle extremely high workload of streaming data in super low latency in many giant tech companies.
+
+Despite its huge success in real-time processing domain, at its deep root, Flink has been faithfully following its inborn philosophy of being [an unified data processing engine for both batch and streaming](https://flink.apache.org/news/2019/02/13/unified-batch-streaming-blink.html), and taking a streaming-first approach in its architecture to do batch processing. By making batch a special case for streaming, Flink really leverages its cutting edge streaming capabilites and apply them to batch scenarios to gain the best offline performance. Flink’s batch performance has been quite outstanding in the early days, and has become even more impressive, as the community began to merge Blink, an Alibaba’s fork of Flink, back to Flink in 1.9 and finished it in 1.10, according to benchmarks we ran. 
+
+On the other hand, Apache Hive has established itself as a focal point of the data warehousing ecosystem. It serves as not only a SQL engine for big data analytics and ETL, but also a data management platform, where data is discovered and defined. As business evolves, it puts new requirements on data warehouse.
+
+Thus we started integrating Flink and Hive as beta release in Flink 1.9. Over the pass few months, we’ve been listening to users’s requests and feedbacks, further entensively enhancing our product, and running rigurous benchmarks (which will be published soon separately). I’m glad to announce that the integration between Flink and Hive is at production grade in [Flink 1.10](https://flink.apache.org/news/2020/02/11/release-1.10.0.html) right now and we can’t wait to walk you through the details.
+
+
+### Compatible with More Hive Versions
+
+The beta version in Flink 1.9 covers Hive 1.2 and 2.3, and rely on Hive’s own backward compatibility for other Hive versions.
+
+In Flink 1.10, we brought full coverage to all Hive versions including 1.0, 1.1, 1.2, 2.0, 2.1, 2.2, 2.3, and 3.1. Take a look [here](https://ci.apache.org/projects/flink/flink-docs-release-1.10/dev/table/hive/#supported-hive-versions).
+
+
+### Unified Metadata Management 
+
+Hive Metastore has evolved into the de facto metadata hub over the years in Hadoop, or even cloud, ecosystem. Many companies have a single Hive Metastore service instance in production to manage all of their schemas, either Hive or non-Hive metadata, as the single source of truth.
+
+In 1.9 we introduced Flink’s [HiveCatalog](https://ci.apache.org/projects/flink/flink-docs-release-1.10/dev/table/hive/hive_catalog.html) , connecting Flink to users’ rich metadata pool. The meaning of `HiveCatalog` has two folds here. First, it enables users to use Hive Metastore to store and manage Flink’s metadata, including tables, UDFs, and statistics of data. Second, it enables Flink to access Hive’s existing metadata, so that Flink can read and write Hive tables.
```

Comment:
```suggestion
In 1.9 we introduced Flink’s [HiveCatalog](https://ci.apache.org/projects/flink/flink-docs-release-1.10/dev/table/hive/hive_catalog.html), connecting Flink to users’ rich metadata pool. The meaning of `HiveCatalog` is two-fold here. First, it allows users of Apache Flink to utilize Hive's Metastore to store and manage Flink’s metadata, including tables, UDFs, and statistics of data. Second, it enables Flink to access Hive’s existing metadata, so that Flink itself can read and write Hive tables.
```